Project-Hotpot/Project-Hotpot/Algorithm/Data Structures/Cache.swift

```diff
@@ -10,12 +10,12 @@ import Foundation
 class Cache: NSObject{
     //TODO: currently using dictionary, so can save to disk easily, but consider alternative implementations?
     var cache: [String: [SongDetails]] = [:]
+    let genreKeys: [String]
     
     init(genreKeys: [String]) {
-        for genreKey in genreKeys {
-            //manually initialize key to empty array
-            cache[genreKey] = []
-        }
+        self.genreKeys = genreKeys
+        super.init()
+        loadCache()
```

Comment:
Does this affect app startup or is there a way to init lazily? cc @toby1236

Reply:
Swift does have the concept of lazy: https://www.hackingwithswift.com/example-code/language/what-are-lazy-variables

It looks like this is synchronously loading from disk. The payloads are so small that the benefit of doing this lazily or in the background is likely unnoticeable to the user.

But it is worth pointing out, it's best practice to do all disk-based operations async. This ensures your UI doesn't stutter as cache and document size increases.